ansible_collections/arista/avd/roles/eos_designs/python_modules/pool_manager/avdpoolmanager.py

```diff
@@ -0,0 +1,168 @@
+# Copyright (c) 2023 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from functools import lru_cache
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+from yaml import SafeDumper, dump, safe_load
+
+from ansible_collections.arista.avd.plugins.filter.natural_sort import natural_sort
+from ansible_collections.arista.avd.plugins.plugin_utils.errors import AristaAvdError
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import get, get_item
+
+if TYPE_CHECKING:
+    from ansible_collections.arista.avd.plugins.plugin_utils.eos_designs_shared_utils import SharedUtils
+
+FILE_HEADER = """\
+# This file is auto-generated by AVD eos_designs.
+# When there is a merge conflict, rebase the branch, accept all current changes, and re-run AVD.
+"""
+
+
+class AvdPoolManager:
```

Comment:
This module would deserve proper pytest given its complexity

Reply:
Done. Please review. I focused on the behavior, and not the individual attributes, so we can refactor/rename anything internal to the class.